src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -508,12 +508,99 @@ namespace AppInstaller::CLI::Workflow
                 return result;
             }
         };
+
+        struct MarketFilter : public details::FilterField
+        {
+            MarketFilter(Manifest::string_t market) : details::FilterField("Market"), m_market(market)
+            {
+                AICLI_LOG(CLI, Verbose, << "Market Comparator created with market: " << m_market);
+            }
+
+            static std::unique_ptr<MarketFilter> Create()
+            {
+                return std::make_unique<MarketFilter>(GetCurrentMarket());
+            }
+
+            InapplicabilityFlags IsApplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                // If both allowed and excluded lists are provided, we only need to check the allowed markets.
+                if (!installer.Markets.AllowedMarkets.empty())
+                {
+                    // Inapplicable if NOT found
+                    if (installer.Markets.AllowedMarkets.end() == std::find(installer.Markets.AllowedMarkets.begin(), installer.Markets.AllowedMarkets.end(), m_market))
+                    {
+                        return InapplicabilityFlags::Market;
+                    }
+                }
+                else if (!installer.Markets.ExcludedMarkets.empty())
+                {
+                    // Inapplicable if found
+                    if (installer.Markets.ExcludedMarkets.end() != std::find(installer.Markets.ExcludedMarkets.begin(), installer.Markets.ExcludedMarkets.end(), m_market))
+                    {
+                        return InapplicabilityFlags::Market;
+                    }
+                }
+
+                return InapplicabilityFlags::None;
+            }
+
+            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                std::string result = "Current market '" + m_market + "' does not match installer markets." +
+                    " Allowed markets: " + GetMarketsListAsString(installer.Markets.AllowedMarkets) +
+                    " Excluded markets: " + GetMarketsListAsString(installer.Markets.ExcludedMarkets);
+                return result;
+            }
+
+        private:
+            Manifest::string_t m_market;
+
+            static Manifest::string_t GetCurrentMarket()

```

Comment:
We already need to get the market for passing to the store; it seems like this should already by a common function (or should have been made so then).  If not, please move to a common location and use by both.